utils/test_results.py

```diff
@@ -1,682 +1,70 @@
-
-    return None
-
-
-def generate_base_query(
+def redis_key(
     repoid: int,
-    ordering: str,
-    ordering_direction: str,
-    should_reverse: bool,
-    branch: str | None,
-    interval_num_days: int,
-    testsuites: list[str] | None = None,
-    term: str | None = None,
-    test_ids: set[str] | None = None,
-) -> TestResultsQuery:
-    term_filter = f"%{term}%" if term else None
-
-    if should_reverse:
-        ordering_direction = "DESC" if ordering_direction == "ASC" else "ASC"
-
-    order_by = f"with_cursor.{ordering} {ordering_direction}, with_cursor.name"
-
-    params: dict[str, int | str | tuple[str, ...] | None] = {
-        "repoid": repoid,
-        "interval": f"{interval_num_days} days",
-        "branch": branch,
-        "test_ids": convert_tuple_else_none(test_ids),
-        "testsuites": convert_tuple_else_none(testsuites),
-        "term": term_filter,
-    }
-
-    filtered_params: dict[str, int | str | tuple[str, ...]] = {
-        k: v for k, v in params.items() if v is not None
-    }
-
-    base_query = f"""
-with
-base_cte as (
-	select rd.*
-	from reports_dailytestrollups rd
-    { "join reports_test rt on rt.id = rd.test_id" if testsuites or term else ""}
-	where
-        rd.repoid = %(repoid)s
-		and rd.date >= current_date - interval %(interval)s
-        { "and rd.branch = %(branch)s" if branch else ""}
-        { "and rd.test_id in %(test_ids)s" if test_ids else ""}
-        { "and rt.testsuite in %(testsuites)s" if testsuites else ""}
-        { "and rt.name like %(term)s" if term else ""}
-),
-failure_rate_cte as (
-	select
-		test_id,
-		CASE
-			WHEN SUM(pass_count) + SUM(fail_count) = 0 THEN 0
-			ELSE SUM(fail_count)::float / (SUM(pass_count) + SUM(fail_count))
-		END as failure_rate,
-		CASE
-			WHEN SUM(pass_count) + SUM(fail_count) = 0 THEN 0
-			ELSE SUM(flaky_fail_count)::float / (SUM(pass_count) + SUM(fail_count))
-		END as flake_rate,
-		MAX(latest_run) as updated_at,
-		AVG(avg_duration_seconds) AS avg_duration,
-        SUM(fail_count) as total_fail_count,
-        SUM(flaky_fail_count) as total_flaky_fail_count,
-        SUM(pass_count) as total_pass_count,
-        SUM(skip_count) as total_skip_count
-	from base_cte
-	group by test_id
-),
-commits_where_fail_cte as (
-	select test_id, array_length((array_agg(distinct unnested_cwf)), 1) as failed_commits_count from (
-		select test_id, commits_where_fail as cwf
-		from base_cte
-		where array_length(commits_where_fail,1) > 0
-	) as tests_with_commits_that_failed, unnest(cwf) as unnested_cwf group by test_id
-),
-last_duration_cte as (
-	select base_cte.test_id, last_duration_seconds from base_cte
-	join (
-		select
-			test_id,
-			max(created_at) as created_at
-		from base_cte
-		group by test_id
-	) as latest_rollups
-    on base_cte.created_at = latest_rollups.created_at
-)
+    branch: str,
+    interval_start: int,
+    interval_end: int | None = None,
+) -> str:
+    key = f"test_results:{repoid}:{branch}:{interval_start}"
```

Comment:
Since this is an int... does this have a reasonable enough level of precision so we actually have some cache hits sometimes?

Like if the timestamp is 1231232132 vs. 1231232135 it seems like that would trigger a new rollup to be cached/created

Reply:
there's no timestamp being use in the key, it's just the range of days that the cached values apply to, for example 1 day, 7 days, or 30 days